core/services/ocr2/plugins/rebalancer/factory.go

```diff
@@ -0,0 +1,50 @@
+package rebalancer
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/rebalancer/liquiditygraph"
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/rebalancer/liquiditymanager"
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/rebalancer/liquidityrebalancer"
+	models2 "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/rebalancer/models"
+)
+
+type PluginFactory struct{}
+
+func NewPluginFactory() *PluginFactory {
+	return &PluginFactory{}
+}
+
+func (p PluginFactory) NewReportingPlugin(config ocr3types.ReportingPluginConfig) (ocr3types.ReportingPlugin[models2.ReportMetadata], ocr3types.ReportingPluginInfo, error) {
+	var offchainConfig models2.PluginConfig
+	if err := json.Unmarshal(config.OffchainConfig, &offchainConfig); err != nil {
+		return nil, ocr3types.ReportingPluginInfo{}, fmt.Errorf("invalid config: %w", err)
+	}
+
+	liquidityRebalancer := liquidityrebalancer.NewDummyRebalancer()
+	liquidityGraph := liquiditygraph.NewGraph()
+	liquidityManagerFactory := liquiditymanager.NewBaseLiquidityManagerFactory()
+
+	return NewPlugin(
+			offchainConfig.F,
```

Comment:
You can get `F` from `config` (ocr3types.ReportingPluginConfig)

```suggestion
			config.F,
```